### PR TITLE
DGUS updates and others

### PR DESCRIPTION
**Pull request by GerogeFu**
The first three commits fix the DGUS bugs. And I want to talk about the fourth commit. When i using SENSORLESS feature , i always failed even with IMPROVE_HOMING_RELIABILITY enabled. And i think we may can backward a bit when homing as the PRUSA did. Then everytime homing succeed according to my test .Open to talk.

### CODE REVIEW

**Comment by thisiskeithb** [comment]
Is https://github.com/MarlinFirmware/Marlin/pull/17287/commits/564e0c7c7184d645b2bf3fc19065aba1debfc885 different than `HOMING_BACKOFF_MM`?

https://github.com/MarlinFirmware/Marlin/blob/1986e1cdf851579958dc1fb245486cb2099582dc/Marlin/Configuration_adv.h#L599

**Comment by GerogeFu** [comment]
> HOMING_BACKOFF_MM

It's different. This will move away before homing. Not after homing.

**Comment by thinkyhead** [comment]
> It's different. This will move away before homing. Not after homing.

If that was your intention, the code is incorrect. It does not back away before homing. Instead, the back-away move that you added occurs after the first homing move hits the end and before the back-off that does the second homing bump (so it is probably hidden inside of that back-off).

Did you test your change both with and without a `HOME_BUMP_MM` distance? Did you test your change with and without `HOMING_BACKOFF_MM` and with the latest code (which includes a 0.2s delay after doing sensorless homing of a CORE axis)?

**Comment by GerogeFu** [comment]
> > It's different. This will move away before homing. Not after homing.
> 
> If that was your intention, the code is incorrect. It does not back away before homing. Instead, the back-away move that you added occurs after the first homing move hits the end and before the back-off that does the second homing bump (so it is probably hidden inside of that back-off).
> 
> Did you test your change both with and without a bump distance? Did you test your change with and without `HOMING_BACKOFF_MM` and with the latest code (which includes a 0.2s delay after doing sensorless homing of a CORE axis)?

Let me check the code again.

**Comment by GerogeFu** [comment]
I change the code. Have a look. And i don't have a chance to test it in DELTA machine , so the code do not take that in consideration. Bump distance do not help when i test 2130 sensorless.

**Comment by thinkyhead** [comment]
Ok, so what happens in your testing if the XY axis is all the way to the far end (e.g., at X300 Y300)? Does the resulting grinding of the axes before homing have any negative effect on sensorless homing?

**Comment by GerogeFu** [comment]
If the XY is already at the far end , it will bump far end if distance is less than HOMING_FIRST_BACKOFF_MM. And i did not find any negative effect at the moment.

**Comment by GerogeFu** [comment]
Is there anything i need to do for this PR?

**Comment by thinkyhead** [comment]
I had to play catch-up and get the config repo in sync with the latest changes here, but if it passes build testing now then I think it's ready to go.

File: Marlin/Configuration_adv.h [hunk 1]
```diff
@@ -592,28 +592,32 @@
...
-#define HOMING_BUMP_DIVISOR { 2, 2, 4 }  // Re-Bump Speed Divisor (Divides the Homing Feedrate)
-//#define QUICK_HOME                     // If homing includes X and Y, do a diagonal move initially
-//#define HOMING_BACKOFF_MM { 2, 2, 2 }  // (mm) Move away from the endstops after homing
+/**
+ * Homing Procedure
+ * Homing (G28) does an indefinite move towards the endstops to establish
+ * the position of the toolhead relative to the workspace.
+ */
+
+//#define SENSORLESS_BACKOFF_MM  { 2, 2 }     // (mm) Backoff from endstops before sensorless homing
```
Comment (JohnnyTheOne) [inline Marlin/Configuration_adv.h:609]:
```suggestion
//#define SENSORLESS_BACKOFF_MM  { 2, 2, 2 }     // (mm) Backoff from endstops before sensorless homing
```

Comment (thinkyhead) [reply Marlin/Configuration_adv.h:609]: I removed the Z element because it is not used and should not be assumed.

**Pull request merged**